libs/ngx-videogular/core/src/lib/services/vg-fullscreen-api/vg-fullscreen-api.service.ts

```diff
@@ -156,27 +156,28 @@ export class VgFullscreenApiService {
           elem = this.medias.toArray()[0].elem;
         }
 
-        this.enterElementInFullScreen(elem);
+        await this.enterElementInFullScreen(elem);
       } else {
-        this.enterElementInFullScreen(this.videogularElement);
+        await this.enterElementInFullScreen(this.videogularElement);
       }
     }
 
     this.isFullscreen = true;
     this.onChangeFullscreen.emit(true);
   }
 
-  enterElementInFullScreen(elem: any) {
-    elem[this.polyfill.request]();
+  async enterElementInFullScreen(elem: any) {
+    await elem[this.polyfill.request]();
   }
 
-  exit() {
+  async exit() {
     this.isFullscreen = false;
     this.onChangeFullscreen.emit(false);
 
-    // Exit from native fullscreen
-    if (this.isAvailable && this.nativeFullscreen && document[this.polyfill.exit]) {
-      document[this.polyfill.exit]();
+    // Check if the document is currently in fullscreen mode
+    if (this.isAvailable && this.nativeFullscreen && document[this.polyfill.element]) {
```

Comment:
apart than a lot of function should be async/await
document[this.polyfill.element] is addition to check if is in fullscreen

Reply:
Shouldn't the check for existence of `polyfill.exit` be left here next to the new check for the fullscreen element? 

Exit fullscreen could still crash on iOS:
https://caniuse.com/mdn-api_document_exitfullscreen

Reply:
Jesus Christ...so there is the possibility that there is an element in fullscreen but we don't have a way to exit??
Safari is really next IE

Reply:
Exactly! With Safari on iOS (not on iPad) one cannot programmatically exit native fullscreen.

Reply:
I again dig dip into sentry logs

[1] https://49ing.sentry.io/issues/4666104481/?environment=PROD&project=1375779&query=&sort=freq&statsPeriod=30d&stream_index=1

I rewrote enterElementInFullScreen to do necessary checks

[2] https://49ing.sentry.io/issues/4666104508/?environment=PROD&project=1375779&sort=freq&statsPeriod=30d&stream_index=2

I rewrote exit to do the necessary checks https://github.com/49ing/ngx-videogular/pull/19/commits/50c76689d490ddf235645f58fc2e001ac9d3fef6

Before I publish new version, please check it out @andreashermann